python/cuml/fil/fil.pyx

```diff
@@ -0,0 +1,225 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import copy
+import ctypes
+import math
+import numpy as np
+import warnings
+
+from numba import cuda
+
+from libcpp cimport bool
+from libc.stdint cimport uintptr_t
+from libc.stdlib cimport calloc, malloc, free
+
+from cuml.common.base import Base
+from cuml.common.handle cimport cumlHandle
+from cuml.utils import get_cudf_column_ptr, get_dev_array_ptr, \
+    input_to_dev_array, zeros
+cimport cuml.common.handle
+cimport cuml.common.cuda
+
+cdef extern from "treelite/c_api.h":
+    ctypedef void* ModelHandle;
+
+cdef extern from "fil/fil.h" namespace "ML::fil":
+
+    cdef enum algo_t:
+        NAIVE,
+        TREE_REORG,
+        BATCH_TREE_REORG
+
+    cdef enum output_t:
+        RAW,
+        PROB,
+        CLASS
+
+    cdef struct dense_node_t:
+        float val
+        int bits
+
+    cdef struct forest:
+        pass
+
+    ctypedef forest* forest_t
+
+    cdef struct forest_params_t:
+        dense_node_t* nodes
+        int depth
+        int ntrees
+        int cols
+        algo_t algo
+        output_t output
+        float threshold
+        pass
+
+    cdef struct treelite_params_t:
+        algo_t algo
+        bool output_class
+        float threshold
+
+    cdef void dense_node_init(dense_node_t*,
+                              float,
+                              float,
+                              int,
+                              bool,
+                              bool)
+
+    cdef void dense_node_decode(dense_node_t*,
+                                float*,
+                                float*,
+                                int*,
+                                bool*,
+                                bool*)
+
+    cdef void init_dense(cumlHandle& handle,
+                         forest_t*,
+                         forest_params_t*)
+
+    cdef void free(cumlHandle& handle,
+                   forest_t)
+
+    cdef void predict(cumlHandle& handle,
+                      forest_t,
+                      float*,
+                      float*,
+                      size_t)
+
+    cdef forest_t from_treelite(cumlHandle& handle,
+                                forest_t*,
+                                ModelHandle,
+                                treelite_params_t*)
+
+cdef class FIL_impl():
+
+    cpdef object handle
+    cdef forest_t* forest_pointer
+    cdef forest_t forest_data
+    cdef ModelHandle* model_ptr
+    cdef object depth
+    cdef object num_trees
+    cdef object algo
+    cdef object output
+    cdef object threshold
+    cdef object n_cols
+    cdef object dtype
+    cdef object verbose
+
+    def __cinit__(self, depth=8, n_estimators=50,
```

Comment:
Only `algo`, `threshold` and `handle` are necessary, all other parameters are not.